docs/conf.py

```diff
@@ -234,7 +234,7 @@ def generate_r_docs(app):
         cmake=3.14.0=h52cb24c_0
     /home/docs/.conda/bin/conda install -q -y -n r_env -c conda-forge \
         r-pkgdown=1.3.0=r351h6115d3f_1000 \
-        r-roxygen2=6.1.1=r35h0357c0b_1001
+        r-roxygen2>=7.0.2
```

Comment:
Is it supposed to be fixed now? Then I should test in at RTD.
BTW, why `>=`? Are LightGBM's docs now incompatible with <7.0.2?

Reply:
yes, it's up to version 7.0.2 on `conda-forge` now. I _believe_ that as of the changes in this PR LightGBM's docs will be compatible with both 6.x and 7.x, but 7.0.2 is the newest release so I thought `<=` made sense just to be sure we don't get older versions.

If you think it's not necessary I'd be happy to revert this change.

Reply:
I just wondered about the current state of compatibility 🙂  I personally interpret `>=` in requirements as incompatibility for older versions. However, I'm OK with either `>=` or complete removal of version specification (as it should be compatible with all versions).

Setting docs test right now to check.

Reply:
Unfortunately it fails. So let's adjust versions for RTD in a separate PR, if you don't mind.

```
[autosummary] generating autosummary for: /home/docs/checkouts/readthedocs.org/user_builds/lightgbm/checkouts/docs/docs/pythonapi/lightgbm.Booster.rst, /home/docs/checkouts/readthedocs.org/user_builds/lightgbm/checkouts/docs/docs/pythonapi/lightgbm.Dataset.rst, /home/docs/checkouts/readthedocs.org/user_builds/lightgbm/checkouts/docs/docs/pythonapi/lightgbm.LGBMClassifier.rst, /home/docs/checkouts/readthedocs.org/user_builds/lightgbm/checkouts/docs/docs/pythonapi/lightgbm.LGBMModel.rst, /home/docs/checkouts/readthedocs.org/user_builds/lightgbm/checkouts/docs/docs/pythonapi/lightgbm.LGBMRanker.rst, /home/docs/checkouts/readthedocs.org/user_builds/lightgbm/checkouts/docs/docs/pythonapi/lightgbm.LGBMRegressor.rst, /home/docs/checkouts/readthedocs.org/user_builds/lightgbm/checkouts/docs/docs/pythonapi/lightgbm.create_tree_digraph.rst, /home/docs/checkouts/readthedocs.org/user_builds/lightgbm/checkouts/docs/docs/pythonapi/lightgbm.cv.rst, /home/docs/checkouts/readthedocs.org/user_builds/lightgbm/checkouts/docs/docs/pythonapi/lightgbm.early_stopping.rst, /home/docs/checkouts/readthedocs.org/user_builds/lightgbm/checkouts/docs/docs/pythonapi/lightgbm.plot_importance.rst, /home/docs/checkouts/readthedocs.org/user_builds/lightgbm/checkouts/docs/docs/pythonapi/lightgbm.plot_metric.rst, /home/docs/checkouts/readthedocs.org/user_builds/lightgbm/checkouts/docs/docs/pythonapi/lightgbm.plot_split_value_histogram.rst, /home/docs/checkouts/readthedocs.org/user_builds/lightgbm/checkouts/docs/docs/pythonapi/lightgbm.plot_tree.rst, /home/docs/checkouts/readthedocs.org/user_builds/lightgbm/checkouts/docs/docs/pythonapi/lightgbm.print_evaluation.rst, /home/docs/checkouts/readthedocs.org/user_builds/lightgbm/checkouts/docs/docs/pythonapi/lightgbm.record_evaluation.rst, /home/docs/checkouts/readthedocs.org/user_builds/lightgbm/checkouts/docs/docs/pythonapi/lightgbm.reset_parameter.rst, /home/docs/checkouts/readthedocs.org/user_builds/lightgbm/checkouts/docs/docs/pythonapi/lightgbm.train.rst

Traceback (most recent call last):
  File "/home/docs/checkouts/readthedocs.org/user_builds/lightgbm/checkouts/docs/docs/conf.py", line 256, in generate_r_docs
    raise RuntimeError(output)
RuntimeError: Collecting package metadata: ...working... done
Solving environment: ...working... done

## Package Plan ##

  environment location: /home/docs/.conda/envs/r_env

  added / updated specs:
    - cmake==3.14.0=h52cb24c_0
    - r-base==3.5.1=h1e0a451_2
    - r-data.table==1.11.4=r351h96ca727_0
    - r-devtools==1.13.6=r351h6115d3f_0
    - r-jsonlite==1.5=r351h96ca727_0
    - r-matrix==1.2_14=r351h96ca727_0
    - r-testthat==2.0.0=r351h29659fb_0


The following packages will be downloaded:

    package                    |            build
    ---------------------------|-----------------
    _libgcc_mutex-0.1          |             main           3 KB
    _r-mutex-1.0.0             |      anacondar_1           2 KB
    binutils_impl_linux-64-2.28.1|       had2808c_3        16.1 MB
    binutils_linux-64-7.2.0    |      had2808c_27           8 KB
    bwidget-1.9.11             |                1         113 KB
    bzip2-1.0.8                |       h7b6447c_0         105 KB
    ca-certificates-2019.11.27 |                0         132 KB
    cairo-1.14.12              |       h8948797_3         1.3 MB
    cmake-3.14.0               |       h52cb24c_0         9.5 MB
    curl-7.67.0                |       hbc83047_0         145 KB
    expat-2.2.6                |       he6710b0_0         187 KB
    fontconfig-2.13.0          |       h9420a91_0         291 KB
    freetype-2.9.1             |       h8a8886c_1         822 KB
    fribidi-1.0.5              |       h7b6447c_0         112 KB
    gcc_impl_linux-64-7.2.0    |       habb00fd_3        72.4 MB
    gcc_linux-64-7.2.0         |      h550dcbe_27           9 KB
    gfortran_impl_linux-64-7.2.0|       hdf63c60_3         8.9 MB
    gfortran_linux-64-7.2.0    |      h550dcbe_27           9 KB
    glib-2.56.2                |       hd408876_0         5.0 MB
    graphite2-1.3.13           |       h23475e2_0         101 KB
    gsl-2.4                    |       h14c3975_4         2.6 MB
    gxx_impl_linux-64-7.2.0    |       hdf63c60_3        18.6 MB
    gxx_linux-64-7.2.0         |      h550dcbe_27           8 KB
    harfbuzz-1.8.8             |       hffaf4a1_0         863 KB
    icu-58.2                   |       h9c2bf20_1        22.5 MB
    jpeg-9b                    |       h024ee3a_2         248 KB
    krb5-1.16.1                |       h173b8e3_7         1.4 MB
    libcurl-7.67.0             |       h20c2e04_0         576 KB
    libgcc-ng-9.1.0            |       hdf63c60_0         8.1 MB
    libgfortran-ng-7.3.0       |       hdf63c60_0         1.3 MB
    libpng-1.6.37              |       hbc83047_0         364 KB
    libssh2-1.8.2              |       h1ba5d50_0         250 KB
    libstdcxx-ng-9.1.0         |       hdf63c60_0         4.0 MB
    libtiff-4.1.0              |       h2733197_0         589 KB
    libuuid-1.0.3              |       h1bed415_2          16 KB
    libxcb-1.13                |       h1bed415_1         502 KB
    libxml2-2.9.9              |       hea5a465_1         2.0 MB
    make-4.2.1                 |       h1bed415_1         429 KB
    openssl-1.1.1d             |       h7b6447c_3         3.7 MB
    pango-1.42.4               |       h049681c_0         528 KB
    pcre-8.43                  |       he6710b0_0         260 KB
    pixman-0.38.0              |       h7b6447c_0         618 KB
    r-assertthat-0.2.0         |   r351h6115d3f_0          57 KB
    r-base-3.5.1               |       h1e0a451_2        39.8 MB
    r-cli-1.0.0                |   r351h6115d3f_1         323 KB
    r-crayon-1.3.4             |   r351h6115d3f_0         745 KB
    r-curl-3.2                 |   r351hadc6856_1         406 KB
    r-data.table-1.11.4        |   r351h96ca727_0         1.4 MB
    r-devtools-1.13.6          |   r351h6115d3f_0         684 KB
    r-digest-0.6.15            |   r351h96ca727_0         150 KB
    r-git2r-0.23.0             |   r351h96ca727_1         863 KB
    r-httr-1.3.1               |   r351h6115d3f_1         437 KB
    r-jsonlite-1.5             |   r351h96ca727_0         1.1 MB
    r-lattice-0.20_35          |   r351h96ca727_0         1.1 MB
    r-magrittr-1.5             |   r351h6115d3f_4         165 KB
    r-matrix-1.2_14            |   r351h96ca727_0         3.7 MB
    r-memoise-1.1.0            |   r351h6115d3f_0          40 KB
    r-mime-0.5                 |   r351h96ca727_0          38 KB
    r-openssl-1.0.2            |   r351h96ca727_1         1.1 MB
    r-praise-1.0.0             |   r351h6115d3f_4          21 KB
    r-r6-2.2.2                 |   r351h6115d3f_0         275 KB
    r-rlang-0.2.1              |   r351h96ca727_0         776 KB
    r-rstudioapi-0.7           |   r351h6115d3f_0         132 KB
    r-testthat-2.0.0           |   r351h29659fb_0         960 KB
    r-whisker-0.3_2            |   r351hf348343_4          66 KB
    r-withr-2.1.2              |   r351h6115d3f_0         161 KB
    rhash-1.3.8                |       h1ba5d50_0         235 KB
    tktable-2.10               |       h14c3975_0          88 KB
    xz-5.2.4                   |       h14c3975_4         366 KB
    zstd-1.3.7                 |       h0b5b093_0         887 KB
    ------------------------------------------------------------
                                           Total:       240.4 MB

The following NEW packages will be INSTALLED:

  _libgcc_mutex      pkgs/main/linux-64::_libgcc_mutex-0.1-main
  _r-mutex           pkgs/r/linux-64::_r-mutex-1.0.0-anacondar_1
  binutils_impl_lin~ pkgs/main/linux-64::binutils_impl_linux-64-2.28.1-had2808c_3
  binutils_linux-64  pkgs/main/linux-64::binutils_linux-64-7.2.0-had2808c_27
  bwidget            pkgs/main/linux-64::bwidget-1.9.11-1
  bzip2              pkgs/main/linux-64::bzip2-1.0.8-h7b6447c_0
  ca-certificates    pkgs/main/linux-64::ca-certificates-2019.11.27-0
  cairo              pkgs/main/linux-64::cairo-1.14.12-h8948797_3
  cmake              pkgs/main/linux-64::cmake-3.14.0-h52cb24c_0
  curl               pkgs/main/linux-64::curl-7.67.0-hbc83047_0
  expat              pkgs/main/linux-64::expat-2.2.6-he6710b0_0
  fontconfig         pkgs/main/linux-64::fontconfig-2.13.0-h9420a91_0
  freetype           pkgs/main/linux-64::freetype-2.9.1-h8a8886c_1
  fribidi            pkgs/main/linux-64::fribidi-1.0.5-h7b6447c_0
  gcc_impl_linux-64  pkgs/main/linux-64::gcc_impl_linux-64-7.2.0-habb00fd_3
  gcc_linux-64       pkgs/main/linux-64::gcc_linux-64-7.2.0-h550dcbe_27
  gfortran_impl_lin~ pkgs/main/linux-64::gfortran_impl_linux-64-7.2.0-hdf63c60_3
  gfortran_linux-64  pkgs/main/linux-64::gfortran_linux-64-7.2.0-h550dcbe_27
  glib               pkgs/main/linux-64::glib-2.56.2-hd408876_0
  graphite2          pkgs/main/linux-64::graphite2-1.3.13-h23475e2_0
  gsl                pkgs/main/linux-64::gsl-2.4-h14c3975_4
  gxx_impl_linux-64  pkgs/main/linux-64::gxx_impl_linux-64-7.2.0-hdf63c60_3
  gxx_linux-64       pkgs/main/linux-64::gxx_linux-64-7.2.0-h550dcbe_27
  harfbuzz           pkgs/main/linux-64::harfbuzz-1.8.8-hffaf4a1_0
  icu                pkgs/main/linux-64::icu-58.2-h9c2bf20_1
  jpeg               pkgs/main/linux-64::jpeg-9b-h024ee3a_2
  krb5               pkgs/main/linux-64::krb5-1.16.1-h173b8e3_7
  libcurl            pkgs/main/linux-64::libcurl-7.67.0-h20c2e04_0
  libedit            pkgs/main/linux-64::libedit-3.1.20181209-hc058e9b_0
  libffi             pkgs/main/linux-64::libffi-3.2.1-hd88cf55_4
  libgcc-ng          pkgs/main/linux-64::libgcc-ng-9.1.0-hdf63c60_0
  libgfortran-ng     pkgs/main/linux-64::libgfortran-ng-7.3.0-hdf63c60_0
  libpng             pkgs/main/linux-64::libpng-1.6.37-hbc83047_0
  libssh2            pkgs/main/linux-64::libssh2-1.8.2-h1ba5d50_0
  libstdcxx-ng       pkgs/main/linux-64::libstdcxx-ng-9.1.0-hdf63c60_0
  libtiff            pkgs/main/linux-64::libtiff-4.1.0-h2733197_0
  libuuid            pkgs/main/linux-64::libuuid-1.0.3-h1bed415_2
  libxcb             pkgs/main/linux-64::libxcb-1.13-h1bed415_1
  libxml2            pkgs/main/linux-64::libxml2-2.9.9-hea5a465_1
  make               pkgs/main/linux-64::make-4.2.1-h1bed415_1
  ncurses            pkgs/main/linux-64::ncurses-6.1-he6710b0_1
  openssl            pkgs/main/linux-64::openssl-1.1.1d-h7b6447c_3
  pango              pkgs/main/linux-64::pango-1.42.4-h049681c_0
  pcre               pkgs/main/linux-64::pcre-8.43-he6710b0_0
  pixman             pkgs/main/linux-64::pixman-0.38.0-h7b6447c_0
  r-assertthat       pkgs/r/linux-64::r-assertthat-0.2.0-r351h6115d3f_0
  r-base             pkgs/r/linux-64::r-base-3.5.1-h1e0a451_2
  r-cli              pkgs/r/linux-64::r-cli-1.0.0-r351h6115d3f_1
  r-crayon           pkgs/r/linux-64::r-crayon-1.3.4-r351h6115d3f_0
  r-curl             pkgs/r/linux-64::r-curl-3.2-r351hadc6856_1
  r-data.table       pkgs/r/linux-64::r-data.table-1.11.4-r351h96ca727_0
  r-devtools         pkgs/r/linux-64::r-devtools-1.13.6-r351h6115d3f_0
  r-digest           pkgs/r/linux-64::r-digest-0.6.15-r351h96ca727_0
  r-git2r            pkgs/r/linux-64::r-git2r-0.23.0-r351h96ca727_1
  r-httr             pkgs/r/linux-64::r-httr-1.3.1-r351h6115d3f_1
  r-jsonlite         pkgs/r/linux-64::r-jsonlite-1.5-r351h96ca727_0
  r-lattice          pkgs/r/linux-64::r-lattice-0.20_35-r351h96ca727_0
  r-magrittr         pkgs/r/linux-64::r-magrittr-1.5-r351h6115d3f_4
  r-matrix           pkgs/r/linux-64::r-matrix-1.2_14-r351h96ca727_0
  r-memoise          pkgs/r/linux-64::r-memoise-1.1.0-r351h6115d3f_0
  r-mime             pkgs/r/linux-64::r-mime-0.5-r351h96ca727_0
  r-openssl          pkgs/r/linux-64::r-openssl-1.0.2-r351h96ca727_1
  r-praise           pkgs/r/linux-64::r-praise-1.0.0-r351h6115d3f_4
  r-r6               pkgs/r/linux-64::r-r6-2.2.2-r351h6115d3f_0
  r-rlang            pkgs/r/linux-64::r-rlang-0.2.1-r351h96ca727_0
  r-rstudioapi       pkgs/r/linux-64::r-rstudioapi-0.7-r351h6115d3f_0
  r-testthat         pkgs/r/linux-64::r-testthat-2.0.0-r351h29659fb_0
  r-whisker          pkgs/r/linux-64::r-whisker-0.3_2-r351hf348343_4
  r-withr            pkgs/r/linux-64::r-withr-2.1.2-r351h6115d3f_0
  readline           pkgs/main/linux-64::readline-7.0-h7b6447c_5
  rhash              pkgs/main/linux-64::rhash-1.3.8-h1ba5d50_0
  tk                 pkgs/main/linux-64::tk-8.6.8-hbc83047_0
  tktable            pkgs/main/linux-64::tktable-2.10-h14c3975_0
  xz                 pkgs/main/linux-64::xz-5.2.4-h14c3975_4
  zlib               pkgs/main/linux-64::zlib-1.2.11-h7b6447c_3
  zstd               pkgs/main/linux-64::zstd-1.3.7-h0b5b093_0


Preparing transaction: ...working... done
Verifying transaction: ...working... done
Executing transaction: ...working... done

Error: package or namespace load failed for ‘pkgdown’ in loadNamespace(j <- i[[1L]], c(lib.loc, .libPaths()), versionCheck = vI[[j]]):
 namespace ‘purrr’ 0.2.5 is being loaded, but >= 0.3.3 is required
Execution halted


During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/docs/checkouts/readthedocs.org/user_builds/lightgbm/envs/docs/lib/python3.7/site-packages/sphinx/cmd/build.py", line 303, in build_main
    args.tags, args.verbosity, args.jobs, args.keep_going)
  File "/home/docs/checkouts/readthedocs.org/user_builds/lightgbm/envs/docs/lib/python3.7/site-packages/sphinx/application.py", line 263, in __init__
    self._init_builder()
  File "/home/docs/checkouts/readthedocs.org/user_builds/lightgbm/envs/docs/lib/python3.7/site-packages/sphinx/application.py", line 325, in _init_builder
    self.emit('builder-inited')
  File "/home/docs/checkouts/readthedocs.org/user_builds/lightgbm/envs/docs/lib/python3.7/site-packages/sphinx/application.py", line 510, in emit
    return self.events.emit(event, self, *args)
  File "/home/docs/checkouts/readthedocs.org/user_builds/lightgbm/envs/docs/lib/python3.7/site-packages/sphinx/events.py", line 80, in emit
    results.append(callback(*args))
  File "/home/docs/checkouts/readthedocs.org/user_builds/lightgbm/checkouts/docs/docs/conf.py", line 260, in generate_r_docs
    raise Exception("An error has occurred while generating documentation for R-package\n" + str(e))
Exception: An error has occurred while generating documentation for R-package
Collecting package metadata: ...working... done
Solving environment: ...working... done
```

Reply:
ah ok, I'll take it out of this PR

Reply:
reverted in https://github.com/microsoft/LightGBM/pull/2530/commits/d98211ff6475b97e77b95e03743c850b00dde055